exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Otel2PrometheusConverter.java

```diff
@@ -449,8 +448,10 @@ private static MetricMetadata convertMetadata(MetricData metricData) {
     String help = metricData.getDescription();
     Unit unit = PrometheusUnitsHelper.convertUnit(metricData.getUnit());
     if (unit != null && !name.endsWith(unit.toString())) {
-      name += "_" + unit;
+      // Need to re-sanitize metric name since unit may contain illegal characters
+      name = sanitizeMetricName(name + "_" + unit);
```

Comment:
same?
```suggestion
      name = name + "_" + sanitizeMetricName(unit);
```

Reply:
No not the same, took a test to find out why. Suppose `name=sample, unit=2`
- `"sample" + sanitizeMetricName(2)` yields `sample__`
- `sanitizeMetricName("sample" + "_" + 2)` yields `sample_2`
The reason is that numbers are not allowed as the first character in metric names, but are valid in subsequent characters.